src/core/ddsi/src/ddsi_deadline.c

```diff
@@ -105,8 +107,23 @@ void ddsi_deadline_renew_instance_real (struct ddsi_deadline_adm *deadline_adm,
      will still be triggered, but has no effect on this instance because in
      the callback the deadline (which will be the updated value) will be
      checked for expiry */
+  ddsrt_mtime_t updatetime = ddsrt_time_monotonic();
+  int64_t deadlines_missed = (updatetime.v - elem->t_last_trigger.v)/deadline_adm->dur;
```

Comment:
Considering the other comment, I think it would make more sense to write this as:
```
if (updatetime.v < elem->t_deadline.v) {
  // update, move entry to end of list
} else {
  // deadline missed
  int64_t deadlines_missed = (updatetime.v - elem->t_deadline.v)/deadline_adm->dur - 1;
  // update, but keep list unchanged
}
```